bindings/netstandard/ElectionGuard/ElectionGuard.Encryption/PrecomputeBufferContext.cs

```diff
@@ -2,6 +2,119 @@
 
 namespace ElectionGuard
 {
+    /// <summary>
+    /// Class that controls the precompute process
+    /// </summary>
+    public class Precompute
+    {
+        static readonly int DummyBufferSize = 100;   // set a buffer size that will be > 0 and < the default of 5000 for initialization
+
+        PrecomputeBufferContext _context;
+
+        /// <summary>
+        /// Default constructor to initialize buffers for testing
+        /// </summary>
+        public Precompute()
+        {
+        }
+
+
+        private PrecomputeStatus _currentStatus = new PrecomputeStatus
+        {
+            Progress = 0,
+            CurrentQueueSize = 0,
+            CurrentState = PrecomputeState.NotStarted
+        };
+        private int _maxBuffers = DummyBufferSize;
+
+        /// <summary>
+        /// Event handler that will give back progress to the calling code
+        /// </summary>
+        public event StatusEventHandler ProgressEvent;
+
+        /// <summary>
+        /// Event handler that will signal that the entire precompute is completed
+        /// </summary>
+        public event StatusEventHandler CompletedEvent;
+
+        /// <summary>
+        /// Internal method used to call the completed event
+        /// </summary>
+        private void OnCompletedEvent()
+        {
+            CompletedEvent?.Invoke(_currentStatus);
+        }
+
+        /// <summary>
+        /// Internal method used to call the progress event
+        /// </summary>
+        private void OnProgressEvent()
+        {
+            ProgressEvent?.Invoke(_currentStatus);
+        }
+
+        /// <summary>
+        /// Get the current status for the current running process
+        /// </summary>
+        /// <returns><see cref="PrecomputeStatus">PrecomputeStatus</see> with all of the latest information</returns>
+        public PrecomputeStatus GetStatus()
+        {
+            return _context.GetStatus();
+        }
+
+        /// <summary>
+        /// Gets the progress of the precompute
+        /// </summary>
+        /// <param name="count">count of the buffer entries</param>
+        /// <param name="queueSize">max size of the buffer queue</param>
+        public void GetProgress(out int count, out int queueSize)
+        {
+            var status = _context.GetStatus();
+            count = (int)status.CurrentQueueSize;
+            queueSize = (int)status.MaxQueueSize;
+        }
+
+        /// <summary>
+        /// Starts the precompute process by creating a new thread to run the process
+        /// </summary>
+        /// <param name="publicKey">The max exponentiation to be calculated</param>
+        /// <param name="buffers">The maximum number of buffers to precompute</param>
+        public void StartPrecomputeAsync(ElementModP publicKey, int buffers = 0)
+        {
+            _currentStatus.CurrentState = PrecomputeState.Running;
+            _context = new PrecomputeBufferContext(publicKey, buffers);
```

Comment:
do you want to hook up the event handlers that are inside the PrecomputeBufferContext? i didnt see that in here anywhere and those events will get swallowed, but maybe thats the expected behavior for this use case